### PR TITLE
FB8-62: Handle Threads_binlog_client with gtid

### PR DESCRIPTION
**Pull request by dutow**
Summary: the original commit for threads_running in 5.6 contained bugs.
The 8.0 port simply ported these bugs to 8.0.
This commit solves them, as 06c075e did for 5.6.

Instead of following the 5.6 implementation, this commit moves the
variable changes inside the mysql_binlog_send function, ensuring
that counters will always be correctly modified when that function
is called.

Squash with D6478040

Reference Patch: https://github.com/facebook/mysql-5.6/commit/06c075e

Test Plan: modified the mtr test case.

### CODE REVIEW

File: mysql-test/t/threads_running.test [hunk 1]
```diff
@@ -1,21 +1 @@
...
-drop table if exists t1, t2;
-create table t1 (a int);
-insert into t1 values (1);
-drop table t1;
---enable_warnings
-FLUSH LOGS;
-
-SHOW STATUS LIKE 'Threads_running';
-SHOW STATUS LIKE 'Threads_binlog_client';
+--source include/threads_running.inc
```
Comment (laurynas-biveinis) [inline mysql-test/t/threads_running.test]: This test should belong to rpl_nogtid suite?

File: mysql-test/t/threads_running_gtid-master.opt [hunk 2]
```diff
@@ -0,0 +1 @@
+--gtid-mode=ON --enforce_gtid_consistency=ON --log_slave_updates=ON
```
Comment (laurynas-biveinis) [inline mysql-test/t/threads_running_gtid-master.opt]: And this one to rpl_gtid suite?

File: mysql-test/t/threads_running.test (hunk 1 above)
Comment (percona-ysorokin) [reply mysql-test/t/threads_running.test]: Agree with @laurynas-biveinis . Please move to `mysql-test/suite/rpl_nogtid/t/threads_running.test`.

File: mysql-test/r/threads_running_gtid.result [hunk 3]
```diff
@@ -0,0 +1,13 @@
+include/master-slave.inc
```
Comment (percona-ysorokin) [inline mysql-test/r/threads_running_gtid.result]: Please move to `mysql-test/suite/rpl_gtid/r/threads_running_gtid.result`

File: mysql-test/r/threads_running.result [hunk 4]
```diff
@@ -3,22 +3,11 @@ Warnings:
 Note	####	Sending passwords in plain text without SSL/TLS is extremely insecure.
 Note	####	Storing MySQL user name or password information in the master info repository is not secure and is therefore not recommended. Please consider using the USER and PASSWORD connection options for START SLAVE; see the 'START SLAVE Syntax' in the MySQL Manual for more information.
 [connection master]
-SHOW STATUS LIKE 'Threads_running';
```
Comment (percona-ysorokin) [inline mysql-test/r/threads_running.result:6]: Please move to `mysql-test/suite/rpl_nogtid/r/threads_running.result`

File: mysql-test/t/threads_running_gtid-master.opt (hunk 2 above)
Comment (percona-ysorokin) [reply mysql-test/t/threads_running_gtid-master.opt]: Please move to `mysql-test/suite/rpl_gtid/t/threads_running-master.opt` and remove `--gtid-mode=ON --enforce_gtid_consistency=ON` options from this file.

File: mysql-test/t/threads_running_gtid-slave.opt [hunk 5]
```diff
@@ -0,0 +1 @@
+--gtid-mode=ON --enforce_gtid_consistency=ON
```
Comment (percona-ysorokin) [inline mysql-test/t/threads_running_gtid-slave.opt]: Remove this file as `--gtid-mode=ON --enforce_gtid_consistency=ON` options will be already present in `my.cnf` of the `rpl_gtid` suite.

File: mysql-test/t/threads_running_gtid.test [hunk 6]
```diff
@@ -0,0 +1 @@
+--source include/threads_running.inc
```
Comment (percona-ysorokin) [inline mysql-test/t/threads_running_gtid.test]: Please move to `mysql-test/suite/rpl_gtid/t/threads_running_gtid.test`

File: sql/rpl_master.cc [hunk 7]
```diff
@@ -1024,9 +1018,17 @@ bool com_binlog_dump_gtid(THD *thd, char *packet, size_t packet_length) {
 
 void mysql_binlog_send(THD *thd, char *log_ident, my_off_t pos,
                        Gtid_set *slave_gtid_executed, uint32 flags) {
+  Global_THD_manager *thd_manager = Global_THD_manager::get_instance();
```
Comment (percona-ysorokin) [inline sql/rpl_master.cc]: `auto* thd_manager = Global_THD_manager::get_instance();`

**Review by percona-ysorokin — CHANGES_REQUESTED** [review]

**Comment by dutow** [comment]
Updated.

**Review by percona-ysorokin — APPROVED** [review]
LGTM

**Review by hermanlee — APPROVED** [review]

**Pull request closed without merging**